addons/mail/static/src/core/web/recipients_input.js

```diff
@@ -0,0 +1,304 @@
+                });
+            },
+        });
+
+        this.openFormViewToEditPartner = useOpenMany2XRecord({
+            fieldString: _t("Edit Partner"),
+            resModel: "res.partner",
+            activeActions: {
+                create: false,
+                createEdit: false,
+                write: true,
+            },
+            /** @param {Record} partner */
+            onRecordSaved: (partner) => {
+                const recipients = this.getAllMailThreadRecipients().filter(
+                    recipient => recipient.partner_id === partner.resId
+                );
+                for (const recipient of recipients) {
+                    Object.assign(recipient, {
+                        email: partner.data.email,
+                        name: partner.data.name,
+                    });
+                }
+            }
+        });
+
+        this.openListViewToSelectResPartner = useSelectCreate({
+            resModel: "res.partner",
+            activeActions: {
+                create: false,
+                link: true, // Unable multi-select
+            },
+            /** @param {Object} resIds */
+            onSelected: async (resIds) => {
+                const partners = await this.orm.searchRead(
+                    "res.partner",
+                    [["id", "in", Array.from(resIds)]],
+                    ["email", "id", "lang", "name"]
+                );
+                for (const partner of partners) {
+                    this.insertAdditionalRecipient({
+                        email: partner.email,
+                        lang: partner.lang,
+                        name: partner.name,
+                        partner_id: partner.id,
+                        persona: { type: "partner", id: partner.id },
+                    });
+                }
+            },
+        });
+    }
+
+    getAutoCompleteSources() {
+        return [
+            {
+                placeholder: _t("Loading..."),
+                /** @param {string} term */
+                options: async (term) => {
+                    const partnerIds = new Set();
+                    const recipients = this.getAllMailThreadRecipients();
+
+                    for (const recipient of recipients) {
+                        if (recipient.partner_id) {
+                            partnerIds.add(recipient.partner_id);
+                        }
+                    }
+
+                    const options = [];
+                    const [name, email] = term ? parseEmail(term) : ["", ""];
+
+                    const limit = 8;
+                    const matches = await this.orm.searchRead(
+                        "res.partner",
+                        [
+                            ["id", "not in", Array.from(partnerIds)],
+                            "|",
+                            ["name", "ilike", term],
+                            ["email", "ilike", term],
```

Comment:
Use email_normalized field. And using "ilike" is probably a bad idea, unless you have a minimum a characters + an '@' (maybe implied by parseEmail, to check).

Reply:
`parseEmail` splits the string provided to it following this rule:
 * if the string is an email, return [email, email]
 * if the string is a mailbox notation return [name, email] (e.g. `name <email>`)
 * if the string is neither return [string, false]

I adapted the domain so that we use the name and email returned by the function. For the last case the part of the domain that needs the email is replaced by a False leaf so that we only check for the name.